gprofiler/utils/node.py

```diff
@@ -0,0 +1,176 @@
+            "expression": f'process.mainModule.require("{os.path.join(module_path, "linux-perf.js")}").start()',
+            "replMode": True,
+        },
+    }
+    _send_socket_request(sock, cdp_request)
+
+
+def _stop_dso(sock: WebSocket, module_path: str) -> None:
+    cdp_request = {
+        "id": 1,
+        "method": "Runtime.evaluate",
+        "params": {
+            "expression": f'process.mainModule.require("{os.path.join(module_path, "linux-perf.js")}").stop()',
+            "replMode": True,
+        },
+    }
+    _send_socket_request(sock, cdp_request)
+
+
+def _copy_module_into_process_ns(process: psutil.Process, musl: bool, version: str) -> str:
+    proc_root = get_proc_root_path(process)
+    libc = "musl" if musl else "glibc"
+    dest_inside_container = get_dest_inside_container(musl, version)
+    dest = resolve_proc_root_links(proc_root, dest_inside_container)
+    if os.path.exists(dest):
+        return dest_inside_container
+    src = resource_path(os.path.join("node", "module", libc, version))
+    shutil.copytree(src, dest)
+    add_permission_dir(dest, stat.S_IROTH, stat.S_IXOTH | stat.S_IROTH)
+    return dest_inside_container
+
+
+def _generate_perf_map(module_path: str) -> None:
+    debugger_url = _get_debugger_url()
+    sock = create_connection(debugger_url)
+    sock.settimeout(2)
+    _load_dso(sock, module_path)
+
+
+def _clean_up(module_path: str, pid: int) -> None:
+    debugger_url = _get_debugger_url()
+    sock = create_connection(debugger_url)
+    _stop_dso(sock, module_path)
+    sock.settimeout(2)
+    os.remove(os.path.join("/tmp", f"perf-{pid}.map"))
+
+
+def get_node_processes() -> List[psutil.Process]:
+    return pgrep_exe(r"(?:^.+/node[^/]*$)")
+
+
+def generate_map_for_node_processes(processes: List[psutil.Process]) -> None:
+    """Iterates over all NodeJS processes, starts debugger for it, finds debugger URL,
+    copies node-linux-perf module into process' namespace, loads module and starts it.
+    After that it links it into gProfiler namespace's /tmp. This lets perf load it"""
+    for process in processes:
+        try:
+            musl = is_musl(process)
+            node_version = get_exe_version(process)
+            node_major_version = node_version[1:].split(".")[0]
+            dest = _copy_module_into_process_ns(process, musl, node_major_version)
+            _start_debugger(process.pid)
+            run_in_ns(["pid", "mnt", "net"], lambda: _generate_perf_map(dest), process.pid, passthrough_exception=True)
+        except Exception as e:
+            logger.warning(f"Could not create debug symbols for pid {process.pid}. Reason: {e}", exc_info=True)
+
+
+def clean_up_node_maps(processes: List[psutil.Process]) -> None:
+    """Stops generating perf maps for each NodeJS process and cleans up generated maps"""
+    for process in processes:
+        try:
+            node_version = get_exe_version(process)
+            node_major_version = node_version[1:].split(".")[0]
```

Comment:
What manipulation are you doing on the version here? (can explain in comment the expected value of `node_version` and the end result you try to get)

Reply:
`v16.3.2` -> `16` is a good such comment IMO